contracts/epochs/FlowEpoch.cdc

```diff
@@ -432,6 +559,97 @@ access(all) contract FlowEpoch {
             FlowEpoch.account.storage.load<Bool>(from: /storage/flowAutomaticRewardsEnabled)
             FlowEpoch.account.storage.save(enabled, to: /storage/flowAutomaticRewardsEnabled)
         }
+        
+        /// Ends the currently active epoch and starts a new one with the provided configuration.
+        /// This function saves the recovery epoch metadata in storage containing the full specification for the new epoch.
+        /// This meta data will be emitted in the EpochRecover service event in the next heart beat interval.
+        /// This function is used within sporks to recover the network from Epoch Fallback Mode (EFM).
+        access(all) fun recoverEpoch(
+            randomSource: String,
+            startView: UInt64,
+            stakingEndView: UInt64,
+            endView: UInt64,
+            targetDuration: UInt64,
+            targetEndTime: UInt64,
+            collectorClusters: [[String]],
+            clusterQCVoteData: [FlowClusterQC.ClusterQCVoteData],
+            dkgPubKeys: [String],
+            nodeIDs: [String])
+        {
+            pre {
+                FlowEpoch.isValidPhaseConfiguration(stakingEndView-startView+1, FlowEpoch.configurableMetadata.numViewsInDKGPhase, endView-startView+1):
```

Comment:
Even though it is only used here and not added, but maybe improve `isValidPhaseConfiguration` and add argument labels, it's very hard to see what the arguments passed to the function are, and if they are correct. They all seem to have the same type too, which makes it very easy to mix them up.

Reply:
https://github.com/onflow/flow-core-contracts/pull/420/commits/5b08984f38e26363d4777233f5fb3077f9f6f406